Docs/pem/revise setting agent privs (#6877)
EnterpriseDB/docs: product_docs/docs/pem/10/managing_pem_agent/setting_agent_privileges.mdx

## Root user versus non-root user

PEM functionality lessens as the privileges of the PEM agent decrease. For complete functionality, run the PEM agent as root. 

> The table below gives a high-level summary of the effects of limniting priveleges. 

Reviewer: ```suggestion
The table below gives a high-level summary of the effects of limiting privileges. 
```

> Further detail is provided in the subsequent sections.

Reviewer: ```suggestion
Further details are provided in the subsequent sections.
```

| Patroni Node Status                             | Linux/Windows    | Requires permission to execute `patronictl`. No data will be returned otherwise. |
| Patroni Cluster Status                          | Linux/Windows    | Requires permission to execute `patronictl`. No data will be returned otherwise. |
| PG HBA Conf                                     | Linux/Windows    | Requires permission to read `pg_hba.conf`. No data will be returned otherwise. |
| Data and Log File Analysis                      | Linux/Windows    | Requires permission to read PGDATA. No data will be returned otherwise. |
| WAL Archive Status                              | Linux/Windows    | Requires read access to the WAL directory. No data will be returned otherwise. |
| Failover Manager Node Status                    | Linux/Windows    | Requires permission to execute `efm`. No data will be returned otherwise. |
| Failover Manager Cluster Info                   | Linux/Windows    | Requires permission to execute `efm`. No data will be returned otherwise. |

### Restarting services

> The Audit Log Manager and Server Log Manager require the PEM agent user to restart the Postgres service to apply changes.

Reviewer: ```suggestion
The Audit Log Manager and Server Log Manager require the PEM agent user to restart the Postgres service for changes to take effect.
```

If the user does not have sufficient privileges (typically only the root user may restart services) these features will not be useable.

### Batch/shell tasks

> On Windows, the PEM Agent cannot run batch tasks unless it the agent user has administrative privileges.

Reviewer: ```suggestion
On Windows, the PEM Agent cannot run batch tasks unless the agent user has administrative privileges.
```

On Linux, the PEM Agent can only run shell tasks if the agent user can become the `batch_script_user` specified in `agent.cfg`.

> This is always true of the root user, and of the `batch_script_user` itself.

Reviewer: ```suggestion
This is always true for the root user and the `batch_script_user`.
```

## Functionality affected by limiting database privileges

> If the PEM agent connects to the monitored database as a non-superuser, the level of functionality will depend on what exactly permissions the agent user has.

Reviewer: ```suggestion
If the PEM agent connects to the monitored database using a non-superuser account, the available functionality will be limited based on the specific privileges granted to that user. 
```

The following sections explain what operations are impacted by Postgres user permissions and what permissions are required for normal operation.

> The PEM agent needs to read data from the `pg_catalog` schema for the majority of SQL-based probes. 

Reviewer: ```suggestion
The PEM agent reads data from the pg_catalog schema for most SQL-based probes. In general, assigning the pg_monitor role to the agent user is sufficient. However, certain catalog functions and probes may require additional privileges beyond pg_monitor.
```

In most cases, the `pg_monitor` role is sufficient, however some catalog functions require additional privileges

> In addition, the agent user must be able to connect to all databases on which it needs to run probes.

Reviewer: ```suggestion
Additionally, the agent user must be able to connect to all target databases where probes need to run. 
```
